cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -373,6 +461,103 @@ def from_proto(cls, proto: v2.device_pb2.DeviceSpecification) -> 'GridDevice':
 
         return GridDevice(metadata)
 
+    def to_proto(
+        self, out: Optional[v2.device_pb2.DeviceSpecification] = None
+    ) -> v2.device_pb2.DeviceSpecification:
+        """Serializes the GridDevice to a DeviceSpecification.
+
+        Args:
+            out: Optional DeviceSpecification to be populated. Fields are populated in-place.
+
+        Returns:
+            The populated DeviceSpecification if out is specified, or the newly created
+            DeviceSpecification.
+        """
+        qubits = self._metadata.qubit_set
+        unordered_pairs = [tuple(pair_set) for pair_set in self._metadata.qubit_pairs]
+        pairs = [
+            (pair[0], pair[1]) if pair[0] <= pair[1] else (pair[1], pair[0])
+            for pair in unordered_pairs
+        ]
+        gateset = self._metadata.gateset
+        gate_durations = self._metadata.gate_durations
+
+        if out is None:
+            out = v2.device_pb2.DeviceSpecification()
+
+        # If fields are already filled (i.e. as part of the old DeviceSpecification format), leave
+        # them as is. Fields populated in the new format do not conflict with how they were
+        # populated in the old format.
+        # TODO(#5050) remove empty checks below once deprecated fields in DeviceSpecification are
+        # removed.
+
+        if len(out.valid_qubits) == 0:
```

Comment:
nit: style guide says use false-y empty lists (https://google.github.io/styleguide/pyguide.html#2144-decision)
```suggestion
        if not out.valid_qubits:
```